internal/entities/quote/common.go

```diff
@@ -44,6 +46,38 @@ func ValidateQuoteHash(hash string) error {
 }
 
 func CalculateCallFee(amount *entities.Wei, config PegConfiguration) *entities.Wei {
-	// TODO implement in GBI-2528
-	return entities.NewWei(100000000000)
+	result := new(entities.Wei)
+
+	percentageFee := calculatePercentageFee(amount, config.GetFeePercentage())
+	result.Add(percentageFee, config.GetFixedFee())
+
+	log.Debugf("Percentage fee: %v%% of %v = %v", config.GetFeePercentage(), amount, percentageFee)
+	log.Debugf("Fixed fee: %v", config.GetFixedFee())
+	log.Debugf("Call fee: %v + %v = %v", percentageFee, config.GetFixedFee(), result)
+	return result
+}
+
+func calculatePercentageFee(amount *entities.Wei, percentage *utils.BigFloat) *entities.Wei {
+	const scale = 1000 // the scale needs to have at least as many zeros as the amount of decimals we want to support in the percentage
```

Comment:
If this is configurable and the LPs can support different number of decimals, this one should be an env var.

Reply:
is not configurable, is 2 decimals as per product requirement